### PR TITLE
Implements EventClass::Execute cases, and functions used by those cases.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/common/globals.cpp [hunk 1]
```diff
@@ -172,6 +176,9 @@ void *g_WakeShapes = nullptr;
 void *g_TurretShapes = nullptr;
 void *g_SamShapes = nullptr;
 void *g_MGunShapes = nullptr;
+int g_FormMove = 0;
```
Comment (tomsons26) [inline src/game/common/globals.cpp]: not sure about the default values here, there are none in original

File: src/game/engine/house.cpp [hunk 2]
```diff
@@ -823,6 +824,93 @@ BOOL HouseClass::Is_Ally(ObjectClass *object) const
     return Is_Ally(object->Owner());
 }
 
+ProdFailType HouseClass::Begin_Production(RTTIType rtti, int id)
```
Comment (tomsons26) [inline src/game/engine/house.cpp]: not sure on the position in the header and cpp for these functions, any specific order they should follow?

File: src/game/engine/house.h [hunk 3]
```diff
@@ -55,6 +55,14 @@ struct ZoneInfo
     int Infantry;
 };
 
+enum ProdFailType
```
Comment (tomsons26) [inline src/game/engine/house.h]: Anything known about this?

File: src/game/engine/building.h [hunk 4]
```diff
@@ -36,6 +36,16 @@ class BuildingClass : public TechnoClass
...
     virtual void Death_Announcement(TechnoClass *killer) const override;
+    virtual FireErrorType Can_Fire(target_t target, WeaponSlotType weapon = WEAPON_SLOT_PRIMARY) const override;
+    virtual target_t Greatest_Threat(ThreatType threat) override;
+    virtual void Assign_Target(target_t target) override;
+    virtual BOOL Captured(HouseClass *house = nullptr) override;
+
+    virtual void Enter_Idle_Mode(BOOL a1 = false) override;
+    virtual void Grand_Opening(int a1);
+    virtual void Update_Buildables();
+    virtual void * Remap_Table();
```
Comment (OmniBlade) [inline src/game/engine/building.h]: Formatting issue with placement of * for the pointer, clang-format will fix it.

File: src/game/engine/foot.cpp [hunk 5]
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
     DEBUG_LOG("***Find_Path_Wrapper exit***\n");
     return real;
 }
+
+void FootClass::Queue_Navigation_List(target_t target)
+{
+    int i;
+    if (target) {
```
Comment (OmniBlade) [inline src/game/engine/foot.cpp]: Code style, you should be explict with conditionals, here it should be `if (target != 0)`.

File: src/game/engine/foot.cpp [hunk 6]
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
...
+        for (i = 0; i < NAV_LENGTH && m_NavList[i]; ++i);
```
Comment (OmniBlade) [inline src/game/engine/foot.cpp]: And again here, `m_NavList[i] != 0`.

File: src/game/engine/foot.cpp [hunk 7]
```diff
@@ -1255,3 +1256,31 @@ PathType *FootClass::Find_Path_Wrapper(cell_t dest, FacingType *buffer, int leng
...
+        if (target != As_Target() || i <= 0) {
+            if (!i) {
```
Comment (OmniBlade) [inline src/game/engine/foot.cpp]: And again here, `i == 0`.

File: src/game/engine/gameevent.h [hunk 8]
```diff
@@ -81,14 +82,14 @@ class GameEventClass
     GameEventClass(GameEventType type, unsigned int gamespeed); // For events: GAMESPEED
     GameEventClass(GameEventType type, HousesType house); // For events: ALLY
     GameEventClass(GameEventType type, RTTIType rtti); // For events: SUSPEND, ABANDON
-    GameEventClass(GameEventType type, RTTIType rtti, unsigned int heap_id); // For events: PLACE, PRODUCE
+    GameEventClass(GameEventType type, RTTIType rtti, int heap_id); // For events: PLACE, PRODUCE
     GameEventClass(GameEventType type, RTTIType rtti, cell_t cell); // For events:
     GameEventClass(GameEventType type, SpecialWeaponType special, cell_t cell); // For events: SPECIAL_PLACE
-    GameEventClass(GameEventType type, void *a2, unsigned long a3); // For events:
+    GameEventClass(GameEventType type, void *a2, unsigned long a3); // For events: ADDPLAYER
```
Comment (OmniBlade) [inline src/game/engine/gameevent.h:88]: Don't use types that don't have a specific size where a specific size is actually required. Expecially don't use long and unsigned long as they are known to have different sizes on different platforms and architectures. Use uint32_t in place of unsigned long, uint16_t in place of unsigned short and uint8_t in place of unsigned char.

File: src/game/engine/gameevent.h [hunk 9]
```diff
@@ -132,93 +135,113 @@ class GameEventClass
...
 
+    // confirmed - FRAMEINFO
     struct FrameEventStruct
     {
         unsigned int m_CRC;
         unsigned short m_CommandCount;
         unsigned char m_Delay;
     };
 
+    // confirmed - TIMING
```
Comment (OmniBlade) [inline src/game/engine/gameevent.h:213]: Are we sure this is a bitfield, looks like two uint16_t's if the bitfield has them as 16 bits?

File: src/game/engine/gameevent.cpp [hunk 10]
```diff
@@ -348,23 +307,35 @@ GameEventClass::GameEventClass(AnimType anim, HousesType owner, coord_t coord) :
...
+    char *messageptr = nullptr;
     EVENT_DEBUG_LOG("GameEventClass::Execute() - Executing '%s' event (GameFrame:%d, EventFrame:%d, ID:%d).\n",
         Name_From(m_Type),
         g_GameFrame,
         m_EventFrame,
         m_HouseID);
 
     switch (m_Type) {
-        case EVENT_ALLY:
+        // done
```
Comment (OmniBlade) [inline src/game/engine/gameevent.cpp]: You don't need to leave comments saying done in code intended to be final, just leave todos on stuff needing doing and everything else is assumed to be done.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Pull request closed without merging**